File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    List<List<Integer>> res = new ArrayList<List<Integer>>();
    public List<List<Integer>> subsets(int[] nums) {
        backtrack(nums, 0, new ArrayList<>());
        return res;
    }

    public void backtrack(int[]nums, int start, List path){
        res.add(new ArrayList<Integer>(path));
        for (int i = start ; i < nums.length ; i++){
            // add
            path.add(nums[i]);
            // recursive
            backtrack(nums, i+1, path);
            // remove
            path.remove(path.size()-1);
        }
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        # [1,2,4,6] => [48,24,12,8]
        i = 0
        res = []
        while(i<len(nums)):
            multiply = 1
            k=0
            while(k<len(nums)):
                if i == k:
                    k+=1
                    continue

                multiply *= nums[k]
                k+=1
            res.append(multiply)
            i+=1

        return res
                

        